src/operations/connect.ts

```diff
@@ -561,12 +554,24 @@ function transformUrlOptions(connStrOptions: any) {
 
   if (connStrOpts.wTimeoutMS) {
     connStrOpts.wtimeout = connStrOpts.wTimeoutMS;
+    connStrOpts.wTimeoutMS = undefined;
   }
 
   if (connStrOptions.srvHost) {
     connStrOpts.srvHost = connStrOptions.srvHost;
   }
 
+  // Any write concern options from the URL will be top-level, so we manually
+  // move them options under `object.writeConcern`
+  const wcKeys = ['w', 'wtimeout', 'j', 'journal', 'fsync'];
```

Comment:
I'd prefer to see this in a const exported by `write_concern.ts`, to make future maintenance easier. Perhaps we can restore the export of `writeConcernKeys` that was deleted and add the missing `journal` key in?